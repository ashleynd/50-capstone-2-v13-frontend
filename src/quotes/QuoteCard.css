/* .QuoteCard {
  max-height: 250px;
  margin-bottom: 1rem;
  color: inherit;
  border-radius: 0;
  border: none;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.card-body {
  font-size: larger;
  padding: 10px;
}
.card-title {
  font-size: 1.6rem;
  padding: 10px;
} */

.author-card {
  margin: auto;
  margin-right: -10px;
  margin-top: 10px;
  margin-bottom: -120px;
  padding: 15px;
  padding-bottom: 10px;
  width: 825px;
  height: 145px;
  /* display: flex; */
  align-items: center;
  justify-content: center;
  background-color: aliceblue;
  border-radius: 5px;
  box-shadow: 8px 8px 15px rgba(0, 0, 0, 0.2);
  background: linear-gradient(to top, #dfe9f3 0%, white 100%);
} 

.author-card-list {
  margin-top: 130px;
  align-items: center;
  justify-content: center;
}

.quote-txt {
  font-size: 1.7rem;
}

.searchbar-quotes {
  margin-bottom: 10px;
}

/* .thoughtBubble3 {
  background: linear-gradient(135deg, #f5f7fa 0%, #c3cfe2 100%);
  width: 285px;
  /* position: relative; */
  /* margin-left: -105px; */
  /* margin-top: 8px; */
  /* margin-bottom: 5px; */
  /* padding: 30px; */
  /* border-radius: 70px; */
  /* box-shadow: 8px 8px 15px rgba(0, 0, 0, 0.2); */
/* } */

/* .thoughtBubble3:before {
content: '';
position: absolute;
width: 0;
height: 0;
bottom: 100%;
left: 4.5em;
border: .75rem solid transparent;
border-top: none;
border-bottom-color: #eff1f7;
filter: drop-shadow(0 -0.0625rem 0.0625rem rgba(0, 0, 0, .1));
} */