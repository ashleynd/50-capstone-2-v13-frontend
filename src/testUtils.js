import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedAQuote = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedAQuote }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
