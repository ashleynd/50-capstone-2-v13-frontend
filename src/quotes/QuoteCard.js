import React, { useContext, useState } from "react";
// import { Container, Row, Col } from "reactstrap";
import "./QuoteCard.css";
// import "./QuoteList.css";
import "./LikeButton.css";
import UserContext from "../auth/UserContext";

/** Show limited information about a quote.
 *
 * Is rendered by QuoteCardList to show a "card" for each quote.
 *
 * Receives like func prop from parent, which is called on like.
 *
 * QuoteCardList -> QuoteCard
 */


function QuoteCard({ id, title, authorName }) {
  // console.debug("QuoteCard");


  const { hasLikedAQuote, LikedQuote } = useContext(UserContext);
  const [isLiked, setisLiked] = useState(true);

  React.useEffect(function updateLikedStatus() {
    // console.debug("QuoteCard useEffect updateLikedStatus", "id=", id);
    setisLiked(hasLikedAQuote(id));
  }, [id, hasLikedAQuote]);

  /** Like a quote */
  async function handleLike(evt) {
    if (hasLikedAQuote(id)) return;
    LikedQuote(id);
    setisLiked(true);
  }



  return (
    <div className="author-card-list">
       {/* <Container>
         <Row xs="4">
          <Col> */}
      <div className="author-card"> {isLiked}
          <h6 className="quote-txt">"{title}"</h6>
          <p><small>{authorName}</small></p>
          <button className={isLiked ? 'Mode-like' : 'Mode-unlike'} onClick={handleLike}>{isLiked ? "Added to Favorites ✅" : "Add to Favorites ❤️"}</button>
        </div>
          {/* </Col>
          </Row>
          </Container> */}
        </div>
  );
}



export default QuoteCard;

