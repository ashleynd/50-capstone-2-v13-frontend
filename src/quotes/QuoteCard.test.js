import React from "react";
import { render } from "@testing-library/react";
import QuoteCard from "./QuoteCard";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  let item = { title: "CEO", salary: 1000000, equity: 10 };
  const { asFragment } = render(
      <UserProvider>
        <QuoteCard item={item} />
      </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

