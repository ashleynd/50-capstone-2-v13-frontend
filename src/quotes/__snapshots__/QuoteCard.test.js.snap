// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="author-card-list"
  >
    <div
      class="author-card"
    >
       
      <h6
        class="quote-txt"
      >
        ""
      </h6>
      <p>
        <small />
      </p>
      <button
        class="Mode-unlike"
      >
        Add to Favorites ❤️
      </button>
    </div>
  </div>
</DocumentFragment>
`;
